src/transformers/models/msclap/modeling_msclap.py

```diff
@@ -0,0 +1,1675 @@
+        return_loss (`bool`, *optional*):
+            Whether or not to return the contrastive loss.
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+# Adapted from https://github.com/microsoft/CLAP/blob/59bc8446e3e9426bf4158810e572b0798a30cf4d/msclap/models/clap.py#L8
+class MSClapProjectionLayer(nn.Module):
+    def __init__(self, config: Union[MSClapAudioConfig, MSClapTextConfig]):
+        super().__init__()
+        self.config = config
+        hidden_size = config.hidden_size
+        projection_dim = config.projection_dim
+
+        self.linear1 = nn.Linear(hidden_size, projection_dim, bias=False)
+        self.linear2 = nn.Linear(projection_dim, projection_dim, bias=False)
+        self.layer_norm = nn.LayerNorm(projection_dim)
+        self.drop = nn.Dropout(config.projection_dropout_prob)
+        self.activation = F.gelu
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        hidden_states1 = self.linear1(hidden_states)
+        hidden_states2 = self.activation(hidden_states1)
+        hidden_states2 = self.linear2(hidden_states2)
+        hidden_states2 = self.drop(hidden_states2)
+
+        hidden_states = self.layer_norm(hidden_states1 + hidden_states2)
+        return hidden_states
+
+
+# Adapted from transformers.models.clap.modeling_clap.ClapPreTrainedModel with Clap->MSClap
+class MSClapPreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = MSClapConfig
+    base_model_prefix = "msclap"
+    supports_gradient_checkpointing = False
+
+    def _init_weights(self, module):
+        """Initialize the weights"""
+        factor = self.config.initializer_factor
+
+        if isinstance(module, nn.Embedding):
+            module.weight.data.normal_(mean=0.0, std=factor * 0.02)
+
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        elif isinstance(module, (nn.Conv2d, nn.Linear)):
+            in_proj_std = (self.config.hidden_size**-0.5) * ((2 * self.config.num_hidden_layers) ** -0.5) * factor
+            nn.init.normal_(module.weight, std=in_proj_std)
+            if module.bias is not None:
+                module.bias.data.zero_()
+
+
+# Adapted from transformers.models.clap.modeling_clap.ClapPreTrainedModel with Clap->MSClap
```

Comment:
I'm under the impression you could add a `Copied from` statement instead ! Either directly in the class level or in the method level